freedomofpress/securedrop
# Bump pyca/cryptography to 2.3 for admin workstation

Versions of cryptography prior to 2.3 have a flaw in the implementation of AES-GCM (https://nvd.nist.gov/vuln/detail/CVE-2018-10903). While ansible/paramiko does not make use of AES-GCM let's still update the dependency.

## Status

Ready for review

## Description of Changes

Partially fixes #3677.

* Bumps pyca/cryptography to 2.3 on admin workstations.
## Testing (Tails)
1. checkout this branch
2. run `./securedrop-admin setup`
3. nspect `.venv/lib/python2.7/site-packages/cryptography/__about__.py` and observe the package version is 2.3
4. `./securedrop-admin install` and observe successful SecureDrop install
5. `./securedrop-admin backup` just to confirm ssh is working properly after install.
## Deployment
Can be manually applied via `./securedrop-admin setup`
The SecureDrop updater gui will automatically upgrade the dependency when 0.9 is released, otherwise `./securedrop-admin setup` is required to pull in the latest version

## Checklist


### If you made changes to `securedrop-admin`:


### If you made non-trivial code changes:
